I-GPU-RFC.md

```diff
@@ -0,0 +1,44 @@
+# RFC: Intel GPU Dialect (I-GPU)
```

Comment:
I do not like the name Intel GPU Dialect. SYCL and L0 do not just GPU devices. The name should be something more generic.

Reply:
since the amd and nvidia dialects are called  AMDGPU and NVGPU dialects, I thought Intel-GPU dialect made sense. We can discuss though. I will add more details about those dialects in the RFC